task 3 done

smallAbyss opened the pull request.

smallAbyss committed:
task 3 done

smallAbyss committed:
task 4 completed

smallAbyss committed:
task4: add getSong

smallAbyss committed:
task4: unsigned -> size_t

smallAbyss committed:
task4: delete "this->", where unneecessary

I'm not sure, is this a good practice

smallAbyss committed:
task 5 done

smallAbyss committed:
task 6: fixed getting fio

cin -> getline to get string with spaces

smallAbyss committed:
task6: in progress

smallAbyss committed:
task6: ascetic snake

(not the final version)

smallAbyss commented:
task 6: fixed getting fio - это task5, только сейчас увидел, а как поменять название коммита не нашёл (если такое вообще возможно)

smallAbyss committed:
task6: colores added

still beta

smallAbyss committed:
task6: done

smallAbyss committed:
task6: some changes

-delete con_field pointer (unused)
- if snake_len == square then endGame

alvls commented:
По ЛР3

__Работа   принята__ |  
-- | --
__Ваш текущий балл:__ |  
Функциональность: | 4,0
Качество   проектирования: | 2,0
Качество кода: | 2,0
Бонус: | 0,0
Штраф: | 0,0
__Итого:__ | 8,0

alvls commented on Luzan_EA/task4/task4.cpp:
```diff
@@ -0,0 +1,433 @@
+class TSongLib {
+private:	
+	vector <TSong> list;
+	vector <Tpair> singersIDList;
```
Поясните смысл ID для песен.

smallAbyss replied:
Если вы про 123 строку, то это ID не песни, а исполнителя. Я предположил, что количество исполнителей будет сильно меньше, поэтому при поиске песни вместо сравнения строк-имён будет немного быстрее сравнивать их ID, который будет даваться на этапе добавления песни
Но с введением бинарного поиска, который ищет сначала по названию (тк песни в первую очередь отсортированы по нему), а потом по исполнителю, это идея, скорее всего, потеряет смысл

А вот причина, по которой этот вектор хранит именно связки ID-Name, а не просто имена - это моя невнимательность. ID - это номер элемента в массиве, так что TPair можно заменить на string и далее внутри других методов заменить присваивание ID на присваивание номера элемента

alvls commented on Luzan_EA/task4/task4.cpp:
```diff
@@ -0,0 +1,433 @@
+		size_t sngrID = 0;
+		size_t tmp_size = singersIDList.size();
+
+		for (size_t i = 1; i < tmp_size; i++) {
+			if (sngrNm == singersIDList[i].Name) {
+				sngrID = singersIDList[i].ID;
+				break;
+			}
+		}
```
Песни хранятся упорядоченно, а ищите вы полным перебором. Непорядок...

smallAbyss replied:
Понял, переделал на бинарный поиск

alvls commented on Luzan_EA/task4/task4.vcxproj:
```diff
@@ -111,6 +111,9 @@
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
+    <ClCompile Include="error.cpp" />
```
Лишний файл в проекте - удалите.

smallAbyss replied:
Сделано

alvls commented on Luzan_EA/task4/task4.vcxproj:
```diff
@@ -111,6 +111,9 @@
   <ItemGroup>
+    <ClCompile Include="error.cpp" />
+    <ClCompile Include="task4.cpp" />
+    <ClCompile Include="test.cpp" />
```
Лишний файл в проекте - удалите.

smallAbyss replied:
Сделано

alvls commented on Luzan_EA/task5/task5.vcxproj:
```diff
@@ -111,6 +111,14 @@
     </Link>
   </ItemDefinitionGroup>
   <ItemGroup>
+    <ClCompile Include="Source.cpp" />
+    <ClCompile Include="check.cpp" />
+    <ClCompile Include="Classes.cpp" />
+    <ClCompile Include="test.cpp" />
+  </ItemGroup>
+  <ItemGroup>
+    <ClInclude Include="Header.h" />
+    <ClInclude Include="Header2.h" />
```
В проекте лишние файлы - удалите

smallAbyss replied:
Сделано

alvls commented:
По ЛР4 исправьте замечания - см. выше.
По ЛР5 все отлично. Удалите лишние файлы из проекта, приму работу.

smallAbyss committed:
delete all unnecessary files

smallAbyss committed:
task4: edited findsong()

ещё поправил методы редактирования, которые не должны редактировать 0 песню (своеобразный   аналог Null)

smallAbyss committed:
task4: edited findSong() - 2

вспомнил про существование singerID

alvls commented:
По ЛР5 я не заметил при прошлом просмотре один странный момент. У вас Касса включает в себя ГЖД, причем столько экземпляров, сколько дней. То есть будут 30 раз скопированы все поезда - так, конечно, не должно быть. Материальные объекты сами по себе, а расписание само по себе. Поэтому баллы такие.

__Работа принята__ |  
-- | --
__Ваш текущий   балл:__ |  
Функциональность: | 5,0
Качество   проектирования: | 1,5
Качество   кода: | 2,5
Бонус: | 0,0
Штраф: | 0,0
__Итого:__ | 9,0

alvls commented:
По ЛР4

__Работа принята__ |  
-- | --
__Ваш текущий   балл:__ |  
Функциональность: | 4,0
Качество   проектирования: | 2,0
Качество   кода: | 2,0
Бонус: | 0,0
Штраф: | 0,0
__Итого:__ | 8,0

alvls commented:
По ЛР6

__Работа принята__ |  
-- | --
__Ваш текущий   балл:__ |  
Функциональность: | 8,0
Качество   проектирования: | 3,5
Качество   кода: | 3,5
Бонус: | 0,0
Штраф: | 0,0
__Итого:__ | 15,0
